### DISCUSSION TITLE
Quando foi sua primeira contribuição no mundo open source?

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#964] [Discussão]
Muita gente acha que precisamos escrever um código fantástico para fazer nossa primeira contribuição no open source e por isso se limitam e não conseguem enviar PRs, issues ou qualquer outra contribuição.

Poderíamos postar nesta issue as nossas primeiras contribuições, para as pessoas se inspirarem e tentarem fazer igual. :pray: 

A ideia da issue veio depois do tweet do @felquis que me relembrou dos meus primeiros passos aqui no GitHub:

[Conversa no Twitter](https://twitter.com/w_oliveiras/status/968860384055758849).

Meu primeiro projeto foi o Pastel, um Grunt Boilerplate para agilizar a criação de sites na agência em que trabalhei pela primeira vez como web dev: https://github.com/woliveiras/pastel/commit/56d7ab3b930e8c032ae4ff46fc4515ca1c045780

Depois eu criei o FEMUG-ABC: https://github.com/femug/femug/issues/70

E daí pra frente nunca mais quis parar de contribuir. :raised_hands: 

Antes disso eu ajudava em foruns de TI e Linux no ORKUT!

**Comment by shadowmaru** [comment]
Dividindo aqui minha experiência:

**Primeiro repositório**: Meu primeiro repositório no GitHub tem quase 10 anos: [CD Covers](https://github.com/shadowmaru/cd-covers) mas não foi muito pra frente. Em tempos de streaming fica até irrelevante. EDIT: como eu dizia, [já tem um projeto sobre isso](https://www.nexojornal.com.br/expresso/2018/02/27/O-arquivo-on-line-de-capas-de-disco-com-800-mil-imagens?utm_term=Autofeed&utm_campaign=Echobox&utm_medium=Social&utm_source=Twitter#link_time=1519834093)

**Primeira gem**: Depois fiz uma gem que até que foi útil para algumas pessoas: o [Microformats Helper](https://github.com/shadowmaru/microformats_helper)

**Projetos brasileiros de terceiros**: Em termos de contribuições em projetos dos outros, ajudei na tradução de dois livros: um sobre Rails 2.1 e o ["Why's (Poignant) Guide to Ruby"](https://github.com/carlosbrando/poignant-br), e fiz algumas coisas no Brazilian Rails

**Projetos internacionais de terceiros**: fiz alguns PRs pro [Spree](https://github.com/spree/spree)

Bom, tudo isso faz 10 anos 😮

**Comment by pedrocss** [comment]
Meu [primeiro PR](https://github.com/galetahub/ckeditor/commit/581691172552b73da494f67479175eac26aa2fd2) foi em 2012 para o projeto [https://github.com/galetahub/ckeditor](https://github.com/galetahub/ckeditor) (Ckeditor integration gem for rails), na época trabalhava como desenvolvedor Ruby on Rails e utilizavamos essa gem em um de nossos projetos na empresa.
 
Já publiquei alguns projetinhos no github, mas nada pretensioso. Gostaria de contribuir mais :)

**Comment by kvnol** [comment]
Meus primeiros PRs e commits [foram em 2013 no Odin](https://github.com/wpbrasil/odin/commits?author=kvnol).

**Comment by felipefialho** [comment]
Bem legal essa issue.

Em tese, minha primeira "contribuição" (em aspas, na verdade foi um experimento), foi o [Cartman em CSS](https://codepen.io/LFeh/pen/qzDCJ) que muita gente deve conhecer, isso aconteceu em meados de 2012.

Com relação a projetos reais, o Github pode me ajudar nisso e percebi que minhas duas primeiras contribuições em repositórios foram em 2013:

- Uma correção para um [repositório de entrevistas de Desenvolvedores Front-end](https://github.com/LFeh?tab=overview&from=2013-04-04)
- Um PR fazendo a conversão do [CSS do Chosen para LESS](https://github.com/harvesthq/chosen/pull/1123)

Pouquissimo tempo depois, [comecei a trabalhar no meu site/blog](https://github.com/LFeh?tab=overview&from=2013-04-15) 😻 

Esses foram os primeiros, mas lembro que o que mais me marcou nesses meus primeiros passos no mundo open-source foi quando uma [modificação que eu fiz no modal do Bootstrap foi aceita](https://github.com/twbs/bootstrap/pull/11162). Fiquei realmente eufórico, porque era um projeto que usava no dia-a-dia e gostava bastante.

**Comment by luizbills** [comment]
Meu primeiro projeto open source, antes mesmo de saber o que era open source, foi uma biblioteca para controlar "buffs" na engine do warcraft 3: https://www.hiveworkshop.com/threads/system-buffgenerator.207342/
Nessa época de fórum eu também já fazia várias contribuição à outros membros.

Já no github:

- meu primeiro projeto foi [um componente](https://github.com/luizbills/CraftySpriteColor) para colorir sprites no framework [Crafty](http://craftyjs.com/) .

- meu primeiro PR foi uma [pequena correção](https://github.com/photonstorm/phaser/pull/63) no framework [PhaserJS](https://github.com/photonstorm/phaser) bem lá na infância dele (hoje o mais famoso framework de jogos em javascript).

> Muita gente acha que precisamos escrever um código fantástico para fazer nossa primeira contribuição no open source e por isso se limitam e não conseguem enviar PRs, issues ou qualquer outra contribuição.

Tem muita gente mesmo que deixa de fazer commit ou PR por achar que é coisa besta ou insignificante. IMHO qualquer ajuda é bem vinda, eu mesmo vivo fazendo PR para corrigir links e erros de ortografia.

**Comment by leandroruel** [comment]
lembrei que estou contribuindo com um agora e faz uma semana que não mexo. vou voltar agora, thanks guys

**Comment by caioincau** [comment]
Minha primeira contribuição foi no VRaptor(framework Java), lá em 2012 :D
